packages/sdk/src/cross-chain-messenger.ts

```diff
@@ -2252,20 +2252,33 @@ export class CrossChainMessenger {
       opts?: {
         recipient?: AddressLike
         l2GasLimit?: NumberLike
-        overrides?: PayableOverrides
+        overrides?: CallOverrides
       },
       isEstimatingGas: boolean = false
     ): Promise<TransactionRequest> => {
       const getOpts = async () => {
         if (isEstimatingGas) {
           return opts
         }
-        const gasEstimation = await this.estimateGas.depositETH(amount, opts)
+        // if we don't include the users address the estimation may fail
+        const from =
+          opts?.overrides?.from ??
+          (ethers.Signer.isSigner(this.l1SignerOrProvider)
+            ? (this.l1SignerOrProvider as Signer).getAddress()
+            : undefined)
```

Comment:
```suggestion
        const from =
          opts?.overrides?.from ??
          (ethers.Signer.isSigner(this.l1SignerOrProvider)
            ? (this.l1SignerOrProvider as Signer).getAddress()
            : undefined)
```
Does ethers not already default to the signer address?

Reply:
it was using the provider instead of signer for gas estimation - I updated the fix to use l1SignerOrProvider instead of l1Provider and cleaned up depositERC20 as well.

Reply:
@roninjin10 if you could take another look at the pr, that would be awesome

Reply:
See my comment here https://github.com/ethereum-optimism/optimism/pull/9949#pullrequestreview-1953937195

Reply:
ok I updated the depositETH test to catch regressions @roninjin10 - the tests now fail without the changes made in this pr